server/service/hosts.go

```diff
@@ -332,14 +332,25 @@ func (svc *Service) DeleteHosts(ctx context.Context, ids []uint, filter *map[str
 		}
 
 		mdmLifecycle := mdmlifecycle.New(svc.ds, svc.logger)
+		lifecycleErrs := []error{}
+		serialsWithErrs := []string{}
 		for _, host := range hosts {
 			if fleet.MDMSupported(host.Platform) {
-				err := mdmLifecycle.Do(ctx, mdmlifecycle.HostOptions{
+				if err := mdmLifecycle.Do(ctx, mdmlifecycle.HostOptions{
 					Action:   mdmlifecycle.HostActionDelete,
 					Host:     host,
 					Platform: host.Platform,
-				})
-				return err
+				}); err != nil {
+					lifecycleErrs = append(lifecycleErrs, err)
+					serialsWithErrs = append(serialsWithErrs, host.HardwareSerial)
+				}
+			}
+		}
+		if len(lifecycleErrs) > 0 {
+			msg := fmt.Sprintf("failed to restore pending host records for one or more MDM devices: %+v", serialsWithErrs)
```

Comment:
Is the message really "restore pending host records"? The lifecycle event is "Delete".

Reply:
I guess it's "restored" since the hosts are still in ABM and will be recreated as pending... Looks ok with that knowledge in mind.

Reply:
I agree it is confusing since it happens during the delete event. Now that you mention it, "recreate" is maybe a better choice of words here so I'll switch it to that.